sros2/test/sros2/commands/security/verbs/test_generate_policy.py

```diff
@@ -36,7 +37,8 @@ def test_generate_policy_topics():
             node.create_publisher(String, 'test_generate_policy_topics_pub', 1)
             node.create_subscription(
                 String, 'test_generate_policy_topics_sub', lambda msg: None, 1)
-
+            # Give time to some rmw implementations to advertise the nodes
+            time.sleep(.500)
             # Generate the policy for the running node
             assert cli.main(
```

Comment:
Looping here until `cli.main(...)` evaluates to `True` or a timeout occurs should be more robust. I believe this is what @hidmic is suggesting. Same for the other test below.

Reply:
Yeah, for some reason now the tests do not pass at all even waiting for more than 1.5 seconds. A better thing to do instead of iterating and waiting will be to use `launch_testing`

Reply:
+1